pkg/state/state.go

```diff
@@ -1902,6 +1902,55 @@ func (s *stateManager) NewestAssetInfo(assetID crypto.Digest) (*proto.AssetInfo,
 	}, nil
 }
 
+func (s *stateManager) NewestFullAssetInfo(assetID crypto.Digest) (*proto.FullAssetInfo, error) {
+	ai, err := s.NewestAssetInfo(assetID)
+	if err != nil {
+		return nil, wrapErr(RetrievalError, err)
+	}
+	info, err := s.stor.assets.newestAssetInfo(assetID, true)
+	if err != nil {
+		return nil, wrapErr(RetrievalError, err)
+	}
+	tx, err := s.TransactionByID(assetID.Bytes())
```

Comment:
We need to use `NewestTransactionByID` here, because this function must take latest changes into account.